### ISSUE TITLE
~/.aws/crendetials & config not updating

### ISSUE THREAD

**Issue by kaburelabs** [vaexio/vaex#1038]
Hi guys, 

I'm using the ~/.aws/credentials & config to set the keys and the region but after setting a new key, id, and region to connect to another s3 account it seems like it still reading the old values. 

How can I force it to change or see where it's getting the values?

One the image below, the values on the credentials aren't related to the loaded table... Why does it happen?
![image](https://user-images.githubusercontent.com/24701735/98030941-870c0300-1df0-11eb-9afa-a3ff6bf60160.png)

**Comment by maartenbreddels** [comment]
Hi Leonardo,


can you try using `'.hdf5?cache=false'`. Are you saying you have the same bucketname, in a different region? I thought that is not possible.

cheers,

Maarten

**Comment by kaburelabs** [comment]
Hey Marteen, thank you for your answer!

I set the ?cache=false argument but the cell keep running for many minutes until I cancel it, what clearly shows that it has something wrong happening 
Error first part
![image](https://user-images.githubusercontent.com/24701735/98155238-ac624500-1eb4-11eb-9d17-d8561e76e690.png)
Error final part
![image](https://user-images.githubusercontent.com/24701735/98155378-e7647880-1eb4-11eb-8d13-4c80d88bc6ee.png)

About the bucket:
No, it's not the same bucket name... It's a different AWS account with a completely new path, key_id, access_key, and region;

I was using a static file in my test, and now I'm trying to use the correct S3 bucket with all data and it's not working after I did change on the ~/.aws/crendetials and config files; 

How can I know if Vaex is using the correct keys or how to force Vaex to use the correct keys?! 

I don't know how to proceed, I'm doing a lot of tests to make it work but it's not being easy;

How can I discover what's happening? 

Thank you in advance for your support!!! 
Leonardo

**Comment by maartenbreddels** [comment]
> I set the ?cache=false argument but the cell keep running for many minutes until I cancel it, what clearly shows that it has something wrong happening

No, that means it is downloading data from s3, which takes a while :)

I still do not understand the problem. You change the while S3 URL right, but you see the same data? Or you have the same S3 URL, but you say it should have different data?

**Comment by maartenbreddels** [comment]
> it's not working after I did change on the ~/.aws/crendetials and config files;

what do you mean by 'not working' ?

**Comment by kaburelabs** [comment]
> > I set the ?cache=false argument but the cell keep running for many minutes until I cancel it, what clearly shows that it has something wrong happening
> 
> No, that means it is downloading data from s3, which takes a while :)
> 
> I still do not understand the problem. You change the while S3 URL right, but you see the same data? Or you have the same S3 URL, but you say it should have different data?

I'm sorry for the messy question Marteen, it's not making sense to me too heheh

I will try to simplify:
I was using a bucket to test my connections with the s3 bucket and now I'm trying to connect to the correct s3 bucket; Now I need to use other S3 URL related to other accounts in a different region to get the correct data; 

 In my understanding, I should only do the modifications on ~/.aws/credentials & config and set the new paths and it should work but It didn't work

> > it's not working after I did change on the ~/.aws/credentials and config files;
> 
> what do you mean by 'not working'?

I think that my confusion is because even with a new key id, access, and region on the credentials & config file, it still loading the old table without loading the new ones. 

Sorry if it still not clear to you... Let me know if I can give you some more information;

**Comment by maartenbreddels** [comment]
> I think that my confusion is because even with a new key id, access, and region on the credentials & config file, it still loading the old table without loading the new ones.

I'm confused by this. The key/acces are just for access rights, if you don't have them correct, you should not be able to access that (please check that, you should get an error from the underlying boto library). But if you have a different S3 URL, what do you mean by 'loading the old table'? Do you mean it's showing the data of the test S3 URL you gave it before?

**Comment by kaburelabs** [comment]
> > I think that my confusion is because even with a new key id, access, and region on the credentials & config file, it still loading the old table without loading the new ones.
> 
> I'm confused by this. The key/access is just for access rights, if you don't have them correct, you should not be able to access that (please check that, you should get an error from the underlying boto library). 

Yeah, I know it and I'm sure that the key/access are correct and it's not returning any boto error 

> But if you have a different S3 URL, what do you mean by 'loading the old table'? Do you mean it's showing the data of the test S3 URL you gave it before?

Perfect, Marteen! Exactly it. Seems like it's based on the file name and not on the path, but it will work only to files where I already had in the test s3 bucket

Take a look at the print below, this file doesn't exist on the s3 bucket but I can load it anyway
![image](https://user-images.githubusercontent.com/24701735/98163337-77a8ba80-1ec1-11eb-88fd-6e4138331140.png)



And when I try to load a that really exists in the s3 bucket I get the error:
`ERROR:MainThread:vaex:error opening 's3://xxxxxx-xxxx-demo-workflows/hdf5/customer_level_table_for_export_2020-09-16 00_00_00.hdf5'`
`....`
`AttributeError: 'int' object has no attribute 'start'`

I'm really stuck on it and It's not making sense; 

I have a hypothesis that I have some problem with the hdf5 file because when I use vaex to convert it returns an hdf5 with 150MB in the other way, it returned a file with ~~58MB... But as It's not giving any error when I load the tables in my local environment... So I really don't know =/

**Comment by kaburelabs** [comment]
Hey @maartenbreddels  
I think that the problem is on the file... I don't know how the guy did the transformation but I did a test using the new hdf5 files and I converted a new .csv to hdf5 file using Vaex that I have never used before and it returns errors only on the hdf5 that aren't converted by Vaex

I will do some more tests to be sure that the problem is this...

Thank you very much for all support bro, I appreciate a lot it;

**Comment by maartenbreddels** [comment]
I'm really surprised by this, the cache is based on the full URL. For instance:
```
$ ls ~/.vaex/file-cache/s3/vaex/testing
xys-masked.hdf5 xys.hdf5
```
So the vaex bucket and other buckets are cached at different locations. Are you doing something with your filesystem maybe?


> `AttributeError: 'int' object has no attribute 'start'`

I think that's unrelated, and fixed by the latest alpha, could you check?
